proc/groupby.go

```diff
@@ -187,38 +190,60 @@ func NewGroupBy(c *Context, parent Proc, params GroupByParams) *GroupBy {
 	// ast.GroupByParams
 	agg := NewGroupByAggregator(c, params)
 	return &GroupBy{
-		Base: Base{Context: c, Parent: parent},
-		agg:  agg,
+		Base:     Base{Context: c, Parent: parent},
+		agg:      agg,
+		resultCh: make(chan Result),
 	}
 }
 
 func (g *GroupBy) Pull() (zbuf.Batch, error) {
-	batch, err := g.Get()
-	if err != nil {
-		return nil, err
-	}
-	if batch == nil {
-		return g.agg.Results(true)
-	}
-	for k := 0; k < batch.Length(); k++ {
-		err := g.agg.Consume(batch.Index(k))
+	g.once.Do(func() { go g.runLoop(g.resultCh) })
+	r := <-g.resultCh
+	return r.Batch, r.Err
+}
+
+func (g *GroupBy) runLoop(ch chan Result) {
```

Comment:
Nit: The corresponding method on both proc.runnerProc and proc.Split is called run, so maybe follow that convention.